pkg/awsutils/awsutils.go

```diff
@@ -918,6 +922,62 @@ func (cache *EC2InstanceMetadataCache) DeallocIPAddresses(eniID string, ips []st
 	return nil
 }
 
+func (cache *EC2InstanceMetadataCache) cleanUpLeakedENIs() {
+	log.Info("Attempting to clean up leaked ENIs")
+
+	networkInterfaces, err := cache.getFilteredListOfNetworkInterfaces()
+	if err != nil {
+		log.Errorf("awsutils: unable to obtain network interfaces.")
+		return
+	}
+
+	for _, networkInterface := range networkInterfaces {
+		// Verify the description starts with "aws-K8S-"
+		if !strings.HasPrefix(aws.StringValue(networkInterface.Description), eniDescriptionPrefix)  {
+			continue
+		}
+		// Delete the ENI.
+		eniID := aws.StringValue(networkInterface.NetworkInterfaceId)
+		err = cache.deleteENI(eniID, maxENIBackoffDelay)
```

Comment:
When this runs on every single node within a large node-group, all but one of these requests would fail. 
Customers might experience throttling issues as well.

Reply:
Status can be either `available` and `in-use`, so yes we do that.

Would probably be better to stagger the get a bit. It's not urgent to clean up the ENIs. Would a random start up delay be enough? Then only the first node would find any ENIs, by the time the next one checks they have already been cleaned up.

Reply:
### Node 1
```
2019-09-25T22:45:40.159Z [INFO]	Will attempt to clean up AWS CNI leaked ENIs after waiting 113 seconds.
.
.
.
2019-09-25T22:47:33.294Z  [DEBUG]	Found 4 available instances with the AWS CNI tag.
2019-09-25T22:47:33.294Z  [DEBUG]	Trying to delete ENI: eni-0e4095e48198b952b
2019-09-25T22:47:33.532Z  [INFO]	Successfully deleted ENI: eni-0e4095e48198b952b
2019-09-25T22:47:33.532Z  [DEBUG]	Trying to delete ENI: eni-07bc73a717a627a08
2019-09-25T22:47:33.797Z  [INFO]	Successfully deleted ENI: eni-07bc73a717a627a08
2019-09-25T22:47:33.797Z  [DEBUG]	Trying to delete ENI: eni-097f5a19626929d9c
2019-09-25T22:47:34.092Z  [INFO]	Successfully deleted ENI: eni-097f5a19626929d9c
2019-09-25T22:47:34.092Z  [DEBUG]	Trying to delete ENI: eni-0630e8e39922861b8
2019-09-25T22:47:34.371Z  [INFO]	Successfully deleted ENI: eni-0630e8e39922861b8
```
### Node 2
```
2019-09-25T22:46:58.173Z  [INFO]	Will attempt to clean up AWS CNI leaked ENIs after waiting 46 seconds.
2019-09-25T22:47:44.323Z  [DEBUG]	No AWS CNI leaked ENIs found.
```

Reply:
I'm seeing that we've implemented our own retry logic and using that inside `deleteENI`. There are 2 problems with that:

- we're unnecessarily re-implementing retry logic in the code. You can simply reuse the k8s wait package which is much more mature and has more functionality - `staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go` (it's already imported it in vendor, so you get it for free :))
- instead of tweaking linear backoff with staggering and increased period, it's better to instead use exponential backoff with jitter for such operations which could interfere with each other (xref - https://stackoverflow.com/a/28732630). Plus this comes for free with the k8s wait pkg